Fix ratelimit issue in token validation

amCap1712 opened the pull request:
Troi using liblistenbrainz was constantly facing more ratelimit errors than expected, and on debugging the issue turned out to be in token validation. The auth token is set on the client after the validate-token endpoint api call, but that means the validate token endpoint itself is subjected to the lower ip based ratelimit. The validate-token endpoint recommends and supports passing the token in the authorization header for the check, and that way the higher ratelimit would be applicable as well. Hence, the change.

amCap1712 committed:
Fix ratelimit issue in token validation

Troi using liblistenbrainz was constantly facing more ratelimit errors than expected,
and on debugging the issue turned out to be in token validation. The auth token is
set on the client after the validate-token endpoint api call, but that means the validate
token endpoint itself is subjected to the lower ip based ratelimit. The validate-token
endpoint recommends and supports passing the token in the authorization header for the
check, and that way the higher ratelimit would be applicable as well. Hence, the change.

mayhem commented on liblistenbrainz/client.py:
```diff
@@ -288,10 +287,7 @@ def is_token_valid(self, token):
         """
-        data = self._get(
-            '/1/validate-token',
-            params={'token': token},
```
I guess this parameter isn't needed?

amCap1712 replied:
remove the method because its not useful then, only way to check validity would be using `set_auth_token` so merged the functionality with it.

amCap1712 committed:
Remove is_token_valid method

To avoid confusion since the token passed is not used anyway and the only way
to test the validity of the token would be by calling set_auth_token in that case.

mayhem reviewed (approved):
Ha, whole function no longer needed.

amCap1712 committed:
update tests
